### PR TITLE
Don't send dev usage report by default

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
Don't send usage reports from unit tests and dev. They are now the majority of usage reports we received.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ci/build/build-multinode-image.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ def build_multinode_image(source_image: str, target_image: str):
     dockerfile = os.path.join(tempdir, "Dockerfile")
     with open(dockerfile, "wt") as f:
         f.write(f"FROM {source_image}\n")
+        f.write("ENV RAY_USAGE_STATS_REPORT_URL=http://127.0.0.1:8000\n")
```
Comment (rkooo567) [inline ci/build/build-multinode-image.py:19]: Is it for building an image? (the env var is not within the container right?)

**Review by rkooo567 — APPROVED** [review]

**Review by ericl — CHANGES_REQUESTED** [review]
Why not? this is easy to filter out and is useful data to analyze.

**Comment by jjyao** [comment]
> Why not? this is easy to filter out and is useful data to analyze.

Yea, it's easy to filter out. Just was not sure if it's useful data for us or just introduce cost with no benefit. But if you think it's something we want to look at, then we can just keep it there.

**Comment by ericl** [comment]
Yes, I'd prefer to keep gathering this data, which is useful as a baseline signal.

**Pull request closed without merging**